libs/stats/odc/stats/model.py

```diff
@@ -400,11 +401,11 @@ def render_assembler_metadata(
                 # when we pass grid, the eodatasets will not load file from path
                 dataset_assembler.note_measurement(band,
                                                     path,
-                                                    pixels=output_dataset[band].values.reshape([self.geobox.shape[0], self.geobox.shape[1]]),
+                                                    pixels=numpy.zeros((self.geobox.shape[0], self.geobox.shape[1])),
```

Comment:
Why does `.note_measurement()` even require a `pixels` argument?

Reply:
In this method, it needs the pixel values to check nodata: https://github.com/GeoscienceAustralia/eo-datasets/blob/3c38a5a740980389452884ba7a53fdee79f7cf1a/eodatasets3/images.py#L308

The calling orders are: 
1. https://github.com/GeoscienceAustralia/eo-datasets/blob/eodatasets3/eodatasets3/assemble.py#L736
2. https://github.com/GeoscienceAustralia/eo-datasets/blob/3c38a5a740980389452884ba7a53fdee79f7cf1a/eodatasets3/images.py#L278
3. https://github.com/GeoscienceAustralia/eo-datasets/blob/3c38a5a740980389452884ba7a53fdee79f7cf1a/eodatasets3/images.py#L297
4. https://github.com/GeoscienceAustralia/eo-datasets/blob/3c38a5a740980389452884ba7a53fdee79f7cf1a/eodatasets3/images.py#L308